x-pack/packages/kbn-elastic-assistant/impl/assistant_context/assistant_nav_link.tsx

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { FC } from 'react';
+import React, { useCallback, useEffect, useMemo } from 'react';
+import ReactDOM from 'react-dom';
+import { createHtmlPortalNode, OutPortal, InPortal } from 'react-reverse-portal';
+import { ChromeNavControls } from '@kbn/core/public';
+import { EuiToolTip, EuiButton, EuiFlexGroup, EuiFlexItem, EuiButtonEmpty } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import useObservable from 'react-use/lib/useObservable';
+import { EMPTY } from 'rxjs';
+import { UseAssistantContext } from '.';
+import { AssistantAvatar } from '../..';
+
+interface Props {
+  hasAssistantPrivilege: UseAssistantContext['assistantAvailability']['hasAssistantPrivilege'];
+  navControls: ChromeNavControls;
+  showAssistantOverlay: UseAssistantContext['showAssistantOverlay'];
+}
+
+const isMac = navigator.platform.toLowerCase().indexOf('mac') >= 0;
+
+const TOOLTIP_CONTENT = i18n.translate(
+  'xpack.elasticAssistant.assistantContext.assistantNavLinkShortcutTooltip',
+  {
+    values: { keyboardShortcut: isMac ? '⌘ ;' : 'Ctrl ;' },
+    defaultMessage: 'Keyboard shortcut {keyboardShortcut}',
+  }
+);
+const LINK_LABEL = i18n.translate('xpack.elasticAssistant.assistantContext.assistantNavLink', {
+  defaultMessage: 'AI Assistant',
+});
+
+export const AssistantNavLink: FC<Props> = ({
+  showAssistantOverlay,
+  hasAssistantPrivilege,
+  navControls,
+}) => {
+  const portalNode = React.useMemo(() => createHtmlPortalNode(), []);
+  const { chrome } = useKibana().services;
+  const chromeStyle$ = useMemo(() => chrome?.getChromeStyle$() ?? EMPTY, [chrome]);
+  const chromeStyle = useObservable(chromeStyle$, undefined);
```

Comment:
The same thing with the `navControls`, we don't need to drill a hole through all those components, we can get it from the context here in this component:

```suggestion
export const AssistantNavLink: FC<Props> = ({
  showAssistantOverlay,
  hasAssistantPrivilege,
}) => {
  const portalNode = React.useMemo(() => createHtmlPortalNode(), []);
  const { chrome } = useKibana().services;
  const  { navControls, getChromeStyle$() } = chrome;
  const chromeStyle$ = useMemo(() => getChromeStyle$(), [getChromeStyle$]);
  const chromeStyle = useObservable(chromeStyle$, undefined);
```